### DISCUSSION TITLE
Exception of Shard 0 already allocated.

### DISCUSSION THREAD

**Discussion by ashuvviet** [akkadotnet/akka.net#5495] [General]
Hi,

we are getting an exception after every re-start of service 3 PODs inside AWS cluster[maxnumbeiof shared =3], we are using Nuget Akka1.4.16 for our .net service using AWS Document DB, attaching the code for your refernce, please let us know if we are missing anything.

Exception in ReceiveRecover when replaying event type [Akka. Cluster.Sharding.PersistentShardCoordinator+ShardHomeAllocated] with sequence number [26] for persistenceId

Cause: System.ArgumentException: Shard 0 is already allocated (Parameter 'e')
   at Akka.Cluster.Sharding.PersistentShardCoordinator.State.Updated(IDomainEvent e)
   at Akka.Cluster.Sharding.PersistentShardCoordinator.ReceiveRecover(Object message)
   at Akka.Persistence.Eventsourced.<>c__DisplayClass91_0.<RecoveryStarted>g__RecoveryBehavior|0(Object message)
   at Akka.Actor.ActorBase.AroundReceive(Receive receive, Object message)
   at Akka.Persistence.Eventsourced.<>n__0(Receive receive, Object message)
   at Akka.Persistence.Eventsourced.<>c__DisplayClass92_0.<Recovering>b__1(Receive receive, Object message)

```
private void ConfigureActorSystem(IServiceCollection services)
        {
            var conf = new Config();
            var setUp = conf.BootstrapApplication(Configuration["ConnectionString"].Replace("{UserName}", Configuration["UserName"]).Replace("{Password}", Configuration["Password"]));
            var dockerconfigsetUp = setUp?.BootstrapFromDocker();
            var bootstrap = BootstrapSetup.Create().WithConfig(dockerconfigsetUp);
            var di = ServiceProviderSetup.Create(services.BuildServiceProvider());
            var actorSystemSetup = bootstrap.And(di);

            var actorSystem = ActorSystem.Create("user-actor-system", actorSystemSetup);
            var shards =3;
            Cluster.Get(actorSystem).RegisterOnMemberUp(() =>
            {
                var provider = Akka.DependencyInjection.ServiceProvider.For(actorSystem);
                var sharding = ClusterSharding.Get(actorSystem);
                var shardRegion = sharding.Start(
                 typeName: nameof(UserActor),
                 entityPropsFactory: s => provider.Props<UserActor>(s), 
                 settings: ClusterShardingSettings.Create(actorSystem),
                 messageExtractor: new MessageExtractor(shards)
              );
                Startup.ShardRegion = shardRegion;

            }
               );
            MongoDbPersistence.Get(actorSystem);
        }
```

**Comment by Aaronontheweb** [comment]
This has been fixed in Akka.NET v1.4.39: https://github.com/akkadotnet/akka.net/releases/tag/1.4.39
